sender_pool_test.go

```diff
@@ -27,111 +27,163 @@ import (
-	s4, err := p.Acquire(ctx)
+	s4, err := p.Sender(ctx)
 	assert.NoError(t, err)
 	assert.Same(t, s1, s4)
 
 	// Acquiring another sender will pop the second one off the slice
-	s5, err := p.Acquire(ctx)
+	s5, err := p.Sender(ctx)
 	assert.NoError(t, err)
 	assert.Same(t, s3, s5)
 }
 
-func TestDoubleReleaseShouldFail(t *testing.T) {
-	p, err := questdb.PoolFromConf("http::addr=localhost:1234")
+func TestFlushOnClose(t *testing.T) {
+	ctx := context.Background()
+
+	srv, err := newTestHttpServer(readAndDiscard)
+	assert.NoError(t, err)
+	defer srv.Close()
+
+	p, err := qdb.PoolFromOptions(
+		qdb.WithHttp(),
+		qdb.WithAddress(srv.Addr()),
+		qdb.WithAutoFlushDisabled(),
+	)
+	assert.NoError(t, err)
+	defer p.Close(ctx)
+
+	s, err := p.Sender(ctx)
+	assert.NoError(t, err)
+
+	err = s.Table(testTable).StringColumn("bar", "baz").AtNow(ctx)
+	assert.NoError(t, err)
+
+	assert.Equal(t, 1, qdb.MsgCount(s))
+
+	assert.NoError(t, s.Close(ctx))
+
+	assert.Equal(t, 0, qdb.MsgCount(s))
+}
+
+func TestPooledSenderDoubleClose(t *testing.T) {
+	p, err := qdb.PoolFromConf("http::addr=localhost:1234")
 	require.NoError(t, err)
 
 	ctx := context.Background()
 
 	// Start with an empty pool, allocate a new sender
-	s1, err := p.Acquire(ctx)
+	s1, err := p.Sender(ctx)
 	assert.NoError(t, err)
 
 	// Release the sender
-	assert.NoError(t, p.Release(ctx, s1))
+	assert.NoError(t, s1.Close(ctx))
 
-	// Try to release the sender again. This should fail because it already exists in the slice
-	assert.Error(t, p.Release(ctx, s1))
+	// Try to release the sender again. This should fail
+	assert.Error(t, s1.Close(ctx))
 }
 
 func TestMaxPoolSize(t *testing.T) {
 	// Create a pool with 2 max senders
-	p, err := questdb.PoolFromConf("http::addr=localhost:1234", questdb.WithMaxSenders(2))
+	p, err := qdb.PoolFromConf("http::addr=localhost:1234", qdb.WithMaxSenders(3))
 	require.NoError(t, err)
 
 	ctx := context.Background()
 
 	// Allocate 3 senders
-	s1, err := p.Acquire(ctx)
+	s1, err := p.Sender(ctx)
 	assert.NoError(t, err)
 
-	s2, err := p.Acquire(ctx)
+	s2, err := p.Sender(ctx)
 	assert.NoError(t, err)
 
-	s3, err := p.Acquire(ctx)
+	s3, err := p.Sender(ctx)
 	assert.NoError(t, err)
 
 	// Release all senders in reverse order
 	// Internal slice will look like: [ s3 , s2 ]
-	assert.NoError(t, p.Release(ctx, s3))
-	assert.NoError(t, p.Release(ctx, s2))
-	assert.NoError(t, p.Release(ctx, s1))
+	assert.NoError(t, s3.Close(ctx))
+	assert.NoError(t, s2.Close(ctx))
+	assert.NoError(t, s1.Close(ctx))
 
 	// Acquire 3 more senders.
 
-	// The first one will be s2 (senders get popped off the slice)
-	s, err := p.Acquire(ctx)
+	// The first one will be s3 (senders get popped off the slice)
+	s, err := p.Sender(ctx)
+	assert.NoError(t, err)
+	assert.Same(t, s, s1)
+
+	// The next will be s2
+	s, err = p.Sender(ctx)
 	assert.NoError(t, err)
 	assert.Same(t, s, s2)
 
-	// The next will be s3
-	s, err = p.Acquire(ctx)
+	// The final one will s1
```

Comment:
Commend is wrong (should be s3)

Reply:
Fixed in 92f05fa913871dd93d23af7775550675719c76c4